### ISSUE TITLE
Support @InjectRepository with multiple connection name

### ISSUE THREAD

**Issue by Dominic-Preap** [nestjs/typeorm#68]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Let's say that I have two mysql servers talking to each other and have a config like below and for some reasons both severs have the same table with name and attributes. So I create only one entity "Photo".
```ts
@Module({
  imports: [
    // ---
    TypeOrmModule.forRootAsync({ useClass: TypeOrmConfigService, name: 'server_one' }),
    TypeOrmModule.forFeature([Photo], 'server_one'),    

    TypeOrmModule.forRootAsync({ useClass: TypeOrmConfigService2, name: 'server_two' }),
    TypeOrmModule.forFeature([Photo], 'server_two')
  ]
})
export class AppModule {}
```
And when I am trying to inject the Photo from "server_two" but I get only Photo from "server_one".

```ts
@Injectable()
export class TypeOrmService {
  constructor(@InjectRepository(Photo)  private readonly photo: Repository<Photo>) {}
}
```

## Expected behavior
Have a second parameter on @InjectRepository(Photo, 'server_name') to specify the server name. 
```ts
@Injectable()
export class TypeOrmService {
  constructor(
    @InjectRepository(Photo, 'server_one')  private readonly photoOne: Repository<Photo>,
    @InjectRepository(Photo, 'server_two')  private readonly photoTwo: Repository<Photo>
  ) {}
}
```

In [typeorm-typedi-extensions](https://github.com/typeorm/typeorm-typedi-extensions#injectrepository), they have this functionality.
> Optionally, you can specify a connection name in the decorator parameters:
> ```ts
> @Service()
> export class PostRepository {
>    
>    @InjectRepository(Post, "custom-con-name")
>    private repository: Repository<Post>;
>    
> }
> ```

## Environment

<pre><code>
Nest version: 5.6.0

 
For Tooling issues:
- Node version: 8.12.0  
- Platform: Windows 

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Would you be interested in creating a PR? :)

**Comment by kamilmysliwiec** [comment]
Published as 6.1.0
